components/zoom_admin/zoom_admin.app.mjs

```diff
@@ -0,0 +1,374 @@
+      optional: true,
+      options: zoomCountries,
+    },
+    pageSize: {
+      type: "integer",
+      label: "Page Size",
+      description: "The number of records returned within a single API call. Defaults to `30`",
+      optional: true,
+      min: 1,
+      max: 300,
+    },
+    pageNumber: {
+      type: "integer",
+      label: "Page Number",
+      description: "The page number of the current page in the returned records.",
+      optional: true,
+      min: 1,
+    },
+    nextPageToken: {
+      type: "string",
+      label: "Next Page Token",
+      description: "The next page token is used to paginate through large result sets. A next page token will be returned whenever the set of available results exceeds the current page size. The expiration period for this token is 15 minutes.",
+      optional: true,
+    },
+    webinar: {
+      type: "string",
+      label: "Webinar",
+      optional: true,
+      description: "The Webinar ID",
+      async options({
+        prevContext,
+        page,
+      }) {
+        if (!prevContext.nextPageToken && page > 0) {
+          return [];
+        }
+
+        const {
+          webinars,
+          next_page_token,
+        } = await this.listWebinars({
+          nextPageToken: prevContext.nextPageToken,
+          pageSize: 2,
+        });
+        if (!webinars.length) {
+          return [];
+        }
+        const rawOptions = webinars.map((w) => ({
+          label: w.topic,
+          value: {
+            label: w.topic,
+            value: w.id,
+          },
+        }));
+        const options = sortBy(rawOptions, [
+          "label",
+        ]);
+
+        return {
+          options,
+          context: {
+            nextPageToken: next_page_token,
+          },
+        };
+      },
+    },
+  },
+  methods: {
+    _apiUrl() {
+      return "https://api.zoom.us/v2";
+    },
+    _getHeaders() {
+      return {
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "user-agent": "@PipedreamHQ/pipedream v0.1",
+      };
+    },
+    _getAxiosParams(opts = {}) {
+      return {
+        ...opts,
+        url: this._apiUrl() + opts.path,
+        headers: this._getHeaders(),
+      };
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers["Accept"] = "application/json";
+      opts.headers["Content-Type"] = "application/json";
+      opts.headers["Authorization"] = `Bearer ${this.$auth.oauth_access_token}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      // eslint-disable-next-line multiline-ternary
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return await axios(opts);
+    },
```

Comment:
Can you refactor this a little bit more so we can
1. Use [pipedream axios](https://pipedream.com/docs/pipedream-axios/#using-pipedream-platform-axios-in-component-actions)
2. Reuse the headers function `_getHeaders` to set necessary headers
3. Centralize all `axios` calls making just use of `_makeRequest` so you don't have to import `axios` on  each isolated component

Reply:
@jcortes As far as I know. It is not possible to use pipedream Axios here because I haven't access to `$` when I am filling the `options`. The `$` variable is only accessible through the `run` function that happens after the `options`. If you have some solution. Please let me know

Reply:
What I normally do is to fallback to `this` if `$` is not present e.g.
```javascript
return await axios($ || this, config);
```